src/app/qgsmaplayerstylecategoriesmodel.cpp

```diff
@@ -68,7 +63,16 @@ QVariant QgsMapLayerStyleCategoriesModel::data( const QModelIndex &index, int ro
   if ( !index.isValid() || index.row() >= rowCount() )
     return QVariant();
 
-  QgsMapLayer::StyleCategory category = index2category( index );
+  QgsMapLayer::StyleCategory category = mCategoryList.at( index.row() + ( mShowAllCategories ? 0 : 1 ) );
```

Comment:
@elpaso sorry, you were right!!!!!

Reply:
I kept your idea of removing index2category and embed this in data. ok for you?

Reply:
Yes, thanks for fixing this!